### PR TITLE
Add our own kodiutils library

### PR DESCRIPTION
**Pull request by dagwieers**
This PR includes:
- Introduce a kodiutils library
- Import some library to blocks where they are needed
- Import only the stuff we need from libraries
- Add proxy-support testing
- Backport xbmc stub libraries

This relates to #94 and #89

### CODE REVIEW

**Comment by dagwieers** [comment]
@mediaminister We test both with and without proxy. In fact all tests run without proxy, but I do an x64 test with proxy. You only need to install proxy.py (`pip install proxy.py` or `pip install -r requirements.txt`)

File: lib/inputstreamhelper/__init__.py [hunk 1]
```diff
@@ -136,7 +93,7 @@ def _diskspace(cls):
     @classmethod
     def _temp_path(cls):
         ''' Return temporary path, usually ~/.kodi/userdata/addon_data/script.module.inputstreamhelper/temp '''
-        temp_path = xbmc.translatePath(os.path.join(ADDON.getSetting('temp_path'), 'temp'))
+        temp_path = xbmc.translatePath(os.path.join(get_setting('temp_path', 'special://masterprofile/addon_data/script.module.inputstreamhelper'), 'temp'))
```
Comment (mediaminister) [inline lib/inputstreamhelper/__init__.py]: I just noticed this bug while testing:
xbmc.translatePath doesn't support unicode in Python2, we need to add `to_unicode()` to support unicode paths in Python 2

```python
to_unicode(xbmc.translatePath(os.path.join(get_setting('temp_path', 'special://masterprofile/addon_data/script.module.inputstreamhelper'), 'temp')))
```

Comment (dagwieers) [reply lib/inputstreamhelper/__init__.py]: Maybe good to add a `translate_path()` function that covers this correctly.

Comment (dagwieers) [reply lib/inputstreamhelper/__init__.py]: I noticed most calls were already doing this.

Comment (dagwieers) [reply lib/inputstreamhelper/__init__.py]: Fixed.

**Comment by mediaminister** [comment]
Looks good to me.

**Pull request merged**